src/plone/restapi/tests/test_blocks_searchable_text.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+from plone.app.testing import setRoles
+from plone.app.testing import SITE_OWNER_NAME
+from plone.app.testing import SITE_OWNER_PASSWORD
+from plone.app.testing import TEST_USER_ID
+from plone.dexterity.interfaces import IDexterityFTI
+from plone.dexterity.utils import createContentInContainer
+from plone.restapi.testing import PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+from plone.restapi.testing import RelativeSession
+from zope.component import queryUtility
+
+import transaction
+import unittest
+
+
+class TestSearchTextInBlocks(unittest.TestCase):
+
+    layer = PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
```

Comment:
Isn't it better to use an integration test here? Or is a functional test mandatory for the restapi? I am not really into it, just asking

Reply:
In plone.restapi tests we usually have to rely on functional tests. You usually do a low-level test setup and then do an API call to check if the expected response is there. This is kind of a functional browser test. We could use integration tests but that would be more work and my personal opinion is that this would lead us to testing on the wrong level. Of course this is debatable for every single use case. Though, we usually stick to functional tests.